satpy/enhancements/contrast.py

```diff
@@ -38,6 +39,30 @@
 LOG = logging.getLogger(__name__)
 
 
+def noop_with_warning(img, msg=""):
+    """Perform no enhancement but emit a warning.
+
+    The warning message is specified by the ``msg`` keyword argument
+    and can be a format string. Keyword arguments will be passed
+    to the formatting of the message string with the underlying
+    ``DataArray``s ``.attrs`` dictionary.
+
+    """
+    formatted_msg = msg.format(**img.data.attrs)
+    warnings.warn(formatted_msg, UserWarning, stacklevel=2)
+    return img
+
+
+def stretch_if_not_integer(img, **kwargs):
+    """Perform a regular linear stretch but warn about no other enhancement."""
+    # XXX: Should this be all integers or just uint8?
+    # XXX: Should it check for "bands" dimension?
+    # XXX: Rename to "stretch_if_necessary"
+    if np.issubdtype(img.data.dtype, np.integer):
+        return img
+    return stretch(img, **kwargs)
```

Comment:
To defend against unusual data (bool, complex, str), I would formulate this the other way around:

```suggestion
if np.issubdtype(img.data.dtype, np.floating):
    return stretch(img, **kwargs)
return img
```